netty/src/main/java/io/grpc/netty/NettyServer.java

```diff
@@ -251,19 +251,10 @@ public void operationComplete(ChannelFuture future) throws Exception {
       throw new IOException("Failed to bind", future.cause());
     }
     channel = future.channel();
-    Future<?> channelzFuture = channel.eventLoop().submit(new Runnable() {
-      @Override
-      public void run() {
-        InternalInstrumented<SocketStats> listenSocket = new ListenSocket(channel);
-        listenSocketStats.set(listenSocket);
-        channelz.addListenSocket(listenSocket);
-      }
-    });
-    try {
-      channelzFuture.await();
-    } catch (InterruptedException ex) {
-      throw new RuntimeException("Interrupted while registering listen socket to channelz", ex);
-    }
+
+    InternalInstrumented<SocketStats> listenSocket = new ListenSocket(channel);
```

Comment:
Make a comment here that this _requires_ the channel to have completed bind()ing.